cmd/mavenBuild.go

```diff
@@ -133,84 +174,96 @@ func runMavenBuild(config *mavenBuildOptions, telemetryData *telemetry.CustomDat
+					deployFlags = append(deployFlags, "-DaltDeploymentRepository="+config.AltDeploymentRepositoryID+"::default::"+config.AltDeploymentRepositoryURL)
+				}
 
-			downloadClient := &piperhttp.Client{}
-			downloadClient.SetOptions(piperhttp.ClientOptions{})
-			runner := &command.Command{
-				StepName: "mavenBuild",
-			}
-			fileUtils := &piperutils.Files{}
-			if len(config.CustomTLSCertificateLinks) > 0 {
-				if err := loadRemoteRepoCertificates(config.CustomTLSCertificateLinks, downloadClient, &deployFlags, runner, fileUtils, config.JavaCaCertFilePath); err != nil {
-					log.SetErrorCategory(log.ErrorInfrastructure)
-					return err
+				downloadClient := &piperhttp.Client{}
+				downloadClient.SetOptions(piperhttp.ClientOptions{})
+
+				runner := &command.Command{
+					StepName: "mavenBuild",
 				}
-			}
 
-			mavenOptions.Flags = deployFlags
-			mavenOptions.Goals = []string{"deploy"}
-			mavenOptions.Defines = []string{}
-			_, err := maven.Execute(&mavenOptions, utils)
-			if err != nil {
-				return err
-			}
-			if config.CreateBuildArtifactsMetadata {
-				buildCoordinates := []versioning.Coordinates{}
-				options := versioning.Options{}
-				var utils versioning.Utils
-
-				matches, _ := fileUtils.Glob("**/pom.xml")
-				for _, match := range matches {
-
-					artifact, err := versioning.GetArtifact("maven", match, &options, utils)
-					if err != nil {
-						log.Entry().Warnf("unable to get artifact metdata : %v", err)
-					} else {
-						coordinate, err := artifact.GetCoordinates()
-						if err != nil {
-							log.Entry().Warnf("unable to get artifact coordinates : %v", err)
-						} else {
-							coordinate.BuildPath = filepath.Dir(match)
-							coordinate.URL = config.AltDeploymentRepositoryURL
-							coordinate.PURL = getPurlForThePomAndDeleteIndividualBom(match)
-							buildCoordinates = append(buildCoordinates, coordinate)
-						}
+				fileUtils := &piperutils.Files{}
+				if len(config.CustomTLSCertificateLinks) > 0 {
+					if err := loadRemoteRepoCertificates(config.CustomTLSCertificateLinks, downloadClient, &deployFlags, runner, fileUtils, config.JavaCaCertFilePath); err != nil {
+						log.SetErrorCategory(log.ErrorInfrastructure)
+						return err
 					}
 				}
 
-				if len(buildCoordinates) == 0 {
-					log.Entry().Warnf("unable to identify artifact coordinates for the maven packages published")
-					return nil
+				mavenOptions.Flags = deployFlags
+				mavenOptions.Goals = []string{"deploy"}
+				mavenOptions.Defines = []string{}
+				_, err = maven.Execute(&mavenOptions, utils)
+				if err != nil {
+					return err
 				}
 
-				var buildArtifacts build.BuildArtifacts
-
-				buildArtifacts.Coordinates = buildCoordinates
-				jsonResult, _ := json.Marshal(buildArtifacts)
-				commonPipelineEnvironment.custom.mavenBuildArtifacts = string(jsonResult)
+				if config.CreateBuildArtifactsMetadata {
+					err2, done := createBuildArtifactsMetadata(config, commonPipelineEnvironment)
```

Comment:
it's totally ok in Go to reuse `err` variable name for all errors unless there is specific need to handle two or more errors at the same time.